packages/annotations/src/format/annotation.js

```diff
@@ -64,19 +60,20 @@ export const annotation = {
 		return null;
 	},
 	__experimentalGetPropsForEditableTreePreparation( select, { richTextIdentifier, blockClientId } ) {
-		return {
-			annotations: select( 'core/annotations' ).__experimentalGetAnnotationsForRichText( blockClientId, richTextIdentifier ),
-		};
+		return select( 'core/annotations' ).__experimentalGetAnnotationsForRichText( blockClientId, richTextIdentifier );
```

Comment:
Is this wrong? the method is called `getProps` I expected it to return an object here?

Reply:
The thing I tried to solve here was not re-rendering unless necessary. `select( 'core/annotations' ).__experimentalGetAnnotationsForRichText( blockClientId, richTextIdentifier )` is cached based on the annotations for a block, so it will only re-render if a different annotation is added. 

With:

```js
return {
	annotations: select( 'core/annotations' ).__experimentalGetAnnotationsForRichText( blockClientId, richTextIdentifier ),
};

it would create a new object every time the function was called, so it would trigger a re-render. And I didn't want to introduce another memoized location, when a simpler solution was available.

Reply:
My hope/thoughts, was that `withSelect` already checks the props (using shallow equality) and we can probably use the same mechanism to do the same.